### PR TITLE
Updating PublishItemsOutputGroup to handle the single file publish scenario

### PR DESCRIPTION
**Pull request by MSLukeWest**
Right now we're using the ResolvedFileToPublish item group, but this doesn't take into account the single file scenario.

### CODE REVIEW

**Comment by nguerrera** [comment]
Adding @swaroop-sridhar

**Comment by nguerrera** [comment]
standard question: for vs 16.5?

**Comment by MSLukeWest** [comment]
> standard question: for vs 16.5?

16.6 is probably good enough.  Is master currently the 16.6 branch?

**Comment by nguerrera** [comment]
It is not, it is the 5.0 release, which will not go into VS for a long while. We don't have a 16.6 branch yet.

**Comment by MSLukeWest** [comment]
> It is not, it is the 5.0 release, which will not go into VS for a long while. We don't have a 16.6 branch yet.

Good to know, thanks.  What's the current target branch for 16.5?  I'll target that for now, then potentially retarget to the 16.6 branch once that exists.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets [hunk 1]
```diff
@@ -1072,17 +1077,24 @@ Copyright (c) .NET Foundation. All rights reserved.
...
-      <PublishItemsOutputGroupOutputs Condition="'$(GenerateDependencyFile)' == 'true' and '$(_UseBuildDependencyFile)' != 'true'"
+      <PublishItemsOutputGroupOutputs Condition="'$(GenerateDependencyFile)' == 'true' and '$(_UseBuildDependencyFile)' != 'true' and '$(PublishSingleFile)' != 'true'"
                                       Include="$(PublishDepsFilePath)"
                                       TargetPath="$(ProjectDepsFileName)" />
       <PublishItemsOutputGroupOutputs Include="@(ResolvedFileToPublish->'%(FullPath)')"
                                       TargetPath="%(ResolvedFileToPublish.RelativePath)"
                                       IsKeyOutput="%(ResolvedFileToPublish.IsKeyOutput)"
+                                      Condition="'$(PublishSingleFile)' != 'true' or '%(ResolvedFileToPublish.ExcludeFromSingleFile)'=='true'"
+                                      OutputGroup="PublishItemsOutputGroup" />
+      <PublishItemsOutputGroupOutputs Include="$(PublishedSingleFilePath)"
```
Comment (swaroop-sridhar) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets]: Why is this step necessary? 
The AppHost executable should already be in `ResolvedFileToPublish`, added by  `_ComputeCopyToPublishDirectoryItems`. 
The AppHost name and the single-file name are the same. So, it shouldn't be necessary to add the singlefile path explicitly.

**Review by swaroop-sridhar — CHANGES_REQUESTED** [review]
Thanks @MSLukeWest  for the fix.
Can you please add a test case to [PublishItemsOutputGroupTests](https://github.com/dotnet/sdk/blob/a5f5bb5183517cb301050288b25c9c727e52e8a6/src/Tests/Microsoft.NET.Publish.Tests/PublishItemsOutputGroupTests.cs)?

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets (hunk 1 above)
Comment (MSLukeWest) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets]: The bundled version of the exe is never added to the ResolvedFileToPublish (or any) item group.  The non-bundled version that is in the item group will now get intentionally excluded in the single file case and the bundled version will be included instead.

**Comment by dsplaisted** [comment]
The 16.5 branch is release/3.1.2xx.  We'll need M2 approval for that branch starting Wednesday.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets (hunk 1 above)
Comment (swaroop-sridhar) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets]: OK the AppHost itself is not marked `ExcludeFromSingleFile`, but handled implicitly; hence the extra explicit copy. Thanks.

**Comment by swaroop-sridhar** [comment]
Looks good to me, other than the request to add a test case. Thanks.

**Comment by MSLukeWest** [comment]
Abandoning this PR, replacing with https://github.com/dotnet/sdk/pull/10600

**Pull request closed without merging**